src/components/app.jsx

```diff
@@ -12,32 +14,76 @@ export default class App extends React.Component {
 
   handleInputChange = (e) => {
     const { value } = e.target;
+    const { contacts } = this.state;
 
-    const filteredContacts = contactsData.filter((contact) => {
+    const filteredContacts = contacts.filter((contact) => {
       const name = contact.name.toLowerCase().replace(/ /g, '');
       return name.includes(value.toLowerCase());
     });
 
+    if (!value) {
+      this.setState({
+        contacts: contactsData,
+      });
+    } else {
+      this.setState({
+        contacts: filteredContacts,
+      });
+    }
+  };
+
+  clearInput = () => {
+    // eslint-disable-next-line no-undef
+    const input = document.getElementById('filter');
+    input.value = '';
+  }
+
+  addNewContact = ({ name, phone }) => {
+    const newID = contactsData.length ? (contactsData[contactsData.length - 1].id + 1) : 0;
```

Comment:
You should work only with contacts stored in state

Reply:
Fixed